crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -121,6 +121,19 @@ def checks
             upgrade_status.end_step
           end
         end
+      rescue ::Crowbar::Error::StartStepRunningError,
+             ::Crowbar::Error::StartStepOrderError,
+             ::Crowbar::Error::SaveUpgradeStatusError => e
+        raise ::Crowbar::Error::UpgradeError.new(e.message)
+      rescue StandardError => e
+        ::Crowbar::UpgradeStatus.new.end_step(
```

Comment:
@skazi0  If you try to start running step again, you should get `Crowbar::Error::StartStepRunningError` and no setting status to failed